scenarioo-server/src/main/java/org/scenarioo/business/diffViewer/ComparisonExecutor.java

```diff
@@ -60,99 +59,183 @@
-					+ baseBuildName + " with defined comparison: " + comparisonConfiguration.getName());
-			} else {
-				comparisonParameters = new ComparisonParameters(baseBranchName, baseBuildName, resolvedComparisonConfiguration,
-					configurationRepository.getConfiguration().getDiffImageAwtColor());
-				storeComparisonInProgress(comparisonParameters);
-				buildDiffInfo = new UseCaseComparator(comparisonParameters).compare();
-				storeComparisonSuccessful(comparisonParameters, buildDiffInfo);
+			BuildDiffInfo buildDiffInfo;
+			ComparisonConfiguration resolvedComparisonConfiguration;
+			ThreadLogAppender comparisonLog = null;
+			ComparisonParameters comparisonParameters = new ComparisonParameters(baseBranchName, baseBuildName, comparisonConfiguration,
+				configurationRepository.getConfiguration().getDiffImageAwtColor());
+			long startTime = System.currentTimeMillis();
+
+			// Precondition check: do nothing if comparison was allready calculated since scheduling it for calculation.
+			try {
+				if (!isComparisonInStateQueuedForComparison(baseBranchName, baseBuildName, comparisonConfiguration.getName())) {
+					// DO NOTHING: The comparison was already processed somehow inbetween (maybe same comparison was scheduled more than once)
+					LOGGER.info("Scheduled Comparsion has allready been calculated inbetween, not calculating it again: "
+						+ getComparisonConfigString(baseBranchName, baseBuildName, comparisonConfiguration));
+					return null;
+				}
+			} catch (Throwable ex) {
+				LOGGER.error("Exception in ComparisonExecutor task in precondition check for comparison: "
+					+ getComparisonConfigString(baseBranchName, baseBuildName, comparisonConfiguration), ex);
 			}
 
-			LOGGER.info("SUCCESS on comparing base build: " + baseBranchName + "/"
-				+ baseBuildName + " with defined comparison: " + comparisonConfiguration.getName());
-			logDuration(startTime);
-			LOGGER.info("=== END OF BUILD COMPARISON (success) ===");
-		} catch (Throwable e) {
-			buildDiffInfo = storeComparisonFailed(comparisonParameters);
-			LOGGER.error("FAILURE on comparing build " + baseBranchName + "/"
-				+ baseBuildName + " with defined comparison: " + comparisonConfiguration.getName(), e);
-			LOGGER.info("=== END OF BUILD COMPARISON (failed) ===");
-		} finally {
-			if (comparisonLog != null) {
-				comparisonLog.unregisterAndFlush();
+
+			try {
+					comparisonLog = registerLogFile(baseBranchName, baseBuildName, comparisonConfiguration);
+					LOGGER.info("=== START OF BUILD COMPARISON ===");
+					LOGGER.info(getComparisonConfigString(baseBranchName, baseBuildName, comparisonConfiguration));
+					resolvedComparisonConfiguration = resolveComparisonConfiguration(comparisonConfiguration, baseBuildName);
+
+					if (resolvedComparisonConfiguration == null) {
+
+						// SKIPPED: Either the builds have not been imported yet, or there is nothing to compare for this comparison
```

Comment:
Extract this block into a separate method

Reply:
ok, did it